src/actions/guides/database/querying.cr

```diff
@@ -7,294 +7,428 @@ class Guides::Database::Querying < GuideAction
-    # The query will run once `each` is called
-    # Results are not cached so a request will be made every time you call `each`
-    query.each do |user|
-      pp user.name
-    end
+    > The `not` method can be used to negate other methods like `eq`, `gt`, `lt`, `in`.
+
+    ### A gt/lt B
+
+    Find rows where `A` is greater than, or equal to, `B`.
+
+    `WHERE users.age >= 21`
+
+    ```crystal
+    UserQuery.new.age.gte(21)
     ```
 
-    ## Finding by ID
+    Find rows where `A` is greater than `B`.
+
+    `WHERE users.created_at > '#{1.day.ago}'`
 
     ```crystal
-    id = 1
-    UserQuery.new.find(id)
+    UserQuery.new.created_at.gt(1.day.ago)
+    ```
+
+    Find rows where `A` is less than, or equal to, `B`.
+
+    `WHERE users.age <= 12`
 
-    # You can also chain methods before calling find, first, etc.
-    UserQuery.new.age(30).find(1)
+    ```crystal
+    UserQuery.new.age.lte(12)
     ```
 
-    ## Ensuring no results are returned
+    Find rows where `A` is less than `B`.
+
+    `WHERE users.updated_at < '#{3.months.ago}'`
 
     ```crystal
-    UserQuery.new.none
+    UserQuery.new.updated_at.lt(3.months.ago)
     ```
 
-    This can be helpful when under certain conditions you want the resultd to be empty.
+    ### A in / not in (B)
 
-    > This method does not return an empty array immediately. You can still chain other query methods, but it
-      will always return no records. For example: `UserQuery.new.none.first` will never return a result
+    Find rows where `A` is in the list `B`.
 
-    ## Shorthand methods
+    `WHERE users.name IN ('Bill', 'John')`
 
-    There are a few methods that are used so often that we've added shortcut methods:
+    ```crystal
+    UserQuery.new.name.in(["Bill", "John"])
+    ```
 
-    * `find`
-    * `first`
-    * `last`
+    Find rows where `A` is not in the list `B`.
+
+    `WHERE users.name NOT IN ('Sally', 'Jenny')`
 
     ```crystal
-    id = 1
-    UserQuery.find(id)
-    UserQuery.first
-    UserQuery.last
+    UserQuery.new.name.not.in(["Sally", "Jenny"])
     ```
 
-    ## Type specific query methods
+    ### A like / iLike B
 
-    The previous section mostly focused on simple equality queries, but you can also
-    do more advanced queries depending on the column type.
+    Find rows where `A` is like (begins with) `B`.
 
-    ### All types
+    `WHERE users.name LIKE 'John%'`
 
-    * `is`
-    * `desc_order`
-    * `asc_order`
+    ```crystal
+    UserQuery.new.name.like("John%")
+    ```
 
-    ### String
+    `WHERE users.name ILIKE 'jim'`
 
-    * `like`
-    * `ilike`
-    * `lower`
+    ```crystal
+    UserQuery.new.name.ilike("jim")
+    ```
 
-    ### Int32 and Time
+    ### Custom WHERE
 
-    * `gt`
-    * `gte`
-    * `lt`
-    * `lte`
+    Avram also has a `where` method that gives you the ability to do custom and/or programmatic queries.
 
-    ### Examples
+    `WHERE users.name = 'Dan' AND created_at > '#{3.days.ago}' AND created_at < '#{1.hour.ago}'`
```

Comment:
I'd be great to get an example here that requires a custom where. For example:

```sql
WHERE name SIMILAR TO '([AEIOU])%'
```

Reply:
good idea

Reply:
I think this is going to get moved to the raw SQL page. I'm adding a note about it.